pynamodb/attributes.py

```diff
@@ -525,6 +535,36 @@ def deserialize(self, value):
         return json.loads(value)
 
 
+class TTLAttribute(Attribute):
+    """
+    An attribute that signifies when how long this item can live for.
+    This can take in an int, timedelta, or datetime.  If a timedelta is passed in,
+    we assume the expiration time is datetime.utcnow() + timedelta.
+    """
+    attr_type = NUMBER
+
+    def serialize(self, value):
+        """
+        Return the epoch representation (in seconds).
+        """
+        if value is None:
+            return None
+        elif isinstance(value, int):
+            return value
+        elif isinstance(value, timedelta):
+            return int(time.time() + value.total_seconds())
+        elif isinstance(value, datetime):
+            return int((value - datetime(1970,1,1)).total_seconds())
+        else:
+            raise ValueError("TTLAttribute value must be an int, timedelta. or datetime.")
+
+    def deserialize(self, value):
+        """
+        Decode the epoch representation into a datetime
+        """
+        return datetime.utcfromtimestamp(value)
```

Comment:
Since UTCDateTimeAttribute returns a tz-aware datetime, maybe
```suggestion
        return datetime.utcfromtimestamp(value).replace(tzinfo=tzutc())
```

Reply:
Didn't think it was necessary to have tz aware datetime but I'll add a tzinfo

Reply:
Just for symmetry with UTCDateTimeAttribute.